pom.xml

```diff
@@ -111,6 +111,31 @@
             <artifactId>auth</artifactId>
             <version>${awssdk.version}</version>
         </dependency>
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>http-auth-spi</artifactId>
```

Comment:
Is there a reason these have to be added

Reply:
These were required on running, without which there was a class not found error when authenticating, with no code changes in my code from previous implementation.
I added the java sdk modules one-by-one until all class not found errors were resolved. This was with the default authentication chain.